platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/internal/PlatformConfigUtils.java

```diff
@@ -102,4 +109,55 @@ private static Set<String> getConfigNames(@NonNull final Configuration configura
                 })
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Write all the settings to the file settingsUsed.txt, some of which might have been changed by settings.txt.
+     *
+     * @param directory the directory to write to
+     */
+    public static void writeSettingsUsed(final Path directory) {
+        final Configuration configuration = ConfigurationHolder.getInstance().get();
+        writeSettingsUsed(directory, configuration);
+    }
+
+    /**
+     * Write all the settings to the file settingsUsed.txt, some of which might have been changed by settings.txt.
+     *
+     * @param directory the directory to write to
+     */
+    public static void writeSettingsUsed(final Path directory, final Configuration configuration) {
+
+        try (final BufferedWriter writer = Files.newBufferedWriter(directory.resolve(SETTING_USED_FILENAME))) {
+            final StringBuilder stringBuilder = new StringBuilder();
+            generateSettingsUsed(stringBuilder, configuration);
+            writer.write(stringBuilder.toString());
+
+            writer.flush();
+        } catch (final IOException e) {
+            logger.error(EXCEPTION.getMarker(), "Error in writing to settingsUsed.txt", e);
+        }
+    }
+
+    /**
+     * Generate the settings used, some of which might have been changed by settings.txt.
+     *
+     * @param stringBuilder the string builder to write to
+     * @param configuration the configuration to use
+     */
+    public static void generateSettingsUsed(final StringBuilder stringBuilder, final Configuration configuration) {
+        stringBuilder.append(PlatformVersion.locateOrDefault().license());
+        stringBuilder.append(System.lineSeparator());
+        stringBuilder.append(System.lineSeparator());
+
+        stringBuilder.append(
+                "The following are all the settings, as modified by settings.txt, but not reflecting any changes "
+                        + "made by config.txt.");
+        stringBuilder.append(System.lineSeparator());
+        stringBuilder.append(System.lineSeparator());
+
+        final Set<String> propertyNames = configuration.getPropertyNames().collect(Collectors.toSet());
+        for (final String propertyName : propertyNames) {
+            stringBuilder.append(String.format("%15s = %s%n", propertyName, configuration.getValue(propertyName)));
+        }
```

Comment:
Storing these in a set could cause the order to be arbitrarily mixed up. Could we rewrite this to guarantee alphabetical order?

Reply:
Used a `TreeSet`